src/librustc/mir/repr.rs

```diff
@@ -850,6 +851,16 @@ pub enum BinOp {
     Gt,
 }
 
+impl BinOp {
+    pub fn is_checkable(self) -> bool {
+        use self::BinOp::*;
+        match self {
+            Add | Sub | Mul | Shl | Shr => true,
```

Comment:
Oh, I totally thought we are using `BinOp` from some lower level, but apparently MIR has its own enumeration.

Never mind me then.